satellite/contact/endpoint.go

```diff
@@ -37,46 +39,69 @@ func NewEndpoint(log *zap.Logger, service *Service) *Endpoint {
 func (endpoint *Endpoint) Checkin(ctx context.Context, req *pb.CheckinRequest) (_ *pb.CheckinResponse, err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	pingNodeSuccess, pingErrorMessage, err := pingBack(ctx, endpoint, req)
+	peerID, err := peerIDFromContext(ctx)
+	pingNodeSuccess, pingErrorMessage, err := pingBack(ctx, endpoint, req, peerID)
 	if err != nil {
 		return nil, Error.Wrap(err)
 	}
 
-	// Update the overlay cache with new uptime info from pinging back the node
-	peerID, err := peerIDFromContext(ctx)
+	// TODO(jg): We are making 2 requests to the database, one to update uptime and
+	// the other to update the capacity and operator info. We should combine these into
+	// one to reduce db connections. Consider adding batching and using a stored procedure.
 	_, err = endpoint.service.overlay.UpdateUptime(ctx, peerID, pingNodeSuccess)
 	if err != nil {
 		return nil, Error.Wrap(err)
 	}
 
-	// Should we update the req.NodeCapacity here as well?
+	nodeInfo := pb.InfoResponse{Operator: req.GetOperator(), Capacity: req.GetCapacity()}
+	_, err = endpoint.service.overlay.UpdateNodeInfo(ctx, peerID, &nodeInfo)
```

Comment:
sounds good, i will add the version field in my next PR, will add it to jira ticket 2571

Reply:
Oh, if we're updating the protos, I have a few small changes to request.
Can we change 'Checkin' to 'CheckIn' since it's two words?
`
service Node {
    rpc Checkin(CheckinRequest) returns (CheckinResponse);
}
`
Could we also change the address field to a string in
`
message CheckinRequest {
    node.NodeAddress address = 1;
    node.NodeCapacity capacity = 3;
    node.NodeOperator operator = 4;
}
`
The node.NodeAddress has an extra Transport field that we don't actually need, but I still have to add it in the request

Reply:
yeah i will add those changes @cam-a